lib/simctl.js

```diff
@@ -345,70 +341,77 @@ async function shutdown (udid) {
- * @param {string} runtimeId - Platform version, for example '10.3'.
+ * @param {string} platformVersion - Platform version, for example '10.3'.
  * @param {?SimCreationOpts} opts - Simulator options for creating devices.
  * @return {string} The UDID of the newly created device.
  * @throws {Error} If the corresponding simctl subcommand command
  *                 returns non-zero return code.
  */
-async function createDevice (name, deviceTypeId, runtimeId, opts = {}) {
+async function createDevice (name, deviceTypeId, platformVersion, opts = {}) {
   const {
     platform = SIM_RUNTIME_NAME_SUFFIX_IOS,
     timeout = DEFAULT_CREATE_SIMULATOR_TIMEOUT
   } = opts;
 
+  let runtimeIds = [];
+
   // Try getting runtimeId using JSON flag
-  let runtimeIdFromJson;
   try {
-    runtimeIdFromJson = await getRuntimeForPlatformVersionViaJson(runtimeId, platform);
-    runtimeId = runtimeIdFromJson;
+    runtimeIds.push(await getRuntimeForPlatformVersionViaJson(platformVersion, platform));
   } catch (ign) { }
 
-  if (!runtimeIdFromJson) {
+  if (_.isEmpty(runtimeIds)) {
     // at first make sure that the runtime id is the right one
     // in some versions of Xcode it will be a patch version
+    let runtimeId;
     try {
-      runtimeId = await getRuntimeForPlatformVersion(runtimeId, platform);
+      runtimeId = await getRuntimeForPlatformVersion(platformVersion, platform);
     } catch (err) {
-      log.warn(`Unable to find runtime for iOS '${runtimeId}'. Continuing`);
+      log.warn(`Unable to find runtime for iOS '${platformVersion}'. Continuing`);
+      runtimeId = platformVersion;
     }
 
-    const clangVersion = await getClangVersion();
-    // 1st comparison: clangVersion
-    // Command line tools version is 10.2+, but xcode 10.1 can happen
-    let isNewerIdFormatRequired = clangVersion && util.compareVersions(clangVersion, '>=',
-      CMDLINE_TOOLS_CLANG_FORMAT_CHANGED_VERSION);
-
-    if (!isNewerIdFormatRequired) {
-      // 2nd comparison: getVersion
-      // The opposite can also happen,
-      // but the combination of 10.2 command line tools and lower Xcode version happens more frequently
-      const xcodeVersion = await getVersion(false);
-      isNewerIdFormatRequired = xcodeVersion && util.compareVersions(xcodeVersion, '>=',
-        XCODE_FORMAT_CHANGED_VERSION);
+    // get the possible runtimes, which will be iterated over
+    // compute the `major.minor` version first since Xcode usually has a runtime
+    // that does not include the patch version, even if it exists
+    const runtimeIdSemver = semver.coerce(runtimeId);
+    if (_.isNil(runtimeIdSemver)) {
+      throw new Error(`Unable to parse runtime id '${runtimeId}'`);
     }
 
-    if (isNewerIdFormatRequired) {
-      runtimeId = `${SIM_RUNTIME_NAME}${platform}-${runtimeId.replace(/\./g, '-')}`;
+    let potentialRuntimeIds = [`${runtimeIdSemver.major}.${runtimeIdSemver.minor}`];
+    if (runtimeId.split('.').length === 3) {
```

Comment:
if (runtimeIdSemver.patch) ...

Reply:
I'm not sure that there is no situation where a `0` patch version is not valid, and it would be excluded here.

Reply:
I'm fine with the current implementation as well